tough/src/error.rs

```diff
@@ -478,6 +496,62 @@ pub enum Error {
 
     #[snafu(display("Role missing from snapshot meta: {}", name))]
     RoleNotInMeta { name: String },
+
+    #[snafu(display("The key for {} was not included", role))]
+    KeyNotFound {
+        role: String,
+        source: schema::Error,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("No keys were found for {}", role))]
+    NoKeys { role: String },
+
+    #[snafu(display("Invalid number"))]
+    InvalidInto {
+        source: std::num::TryFromIntError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Invalid threshold number"))]
+    InvalidThreshold { backtrace: Backtrace },
+
+    /// The library failed to serialize an object to JSON.
+    #[snafu(display("Failed to serialize to JSON: {}", source))]
+    JsonSerialization {
+        source: schema::Error,
+        backtrace: Backtrace,
+    },
+
+    /// Invalid path permissions
+    #[snafu(display("Invalid path permission of {} : {:?}", name, paths))]
+    InvalidPathPermission {
+        name: String,
+        paths: Vec<String>,
+        source: schema::Error,
+    },
+
+    /// Overflow error with checked add
+    #[snafu(display("Overflow of checked add"))]
```

Comment:
I'm not sure what this error message means. Could we clarify?

Reply:
This error is thrown whenever a checked add fails. Maybe `Checked add failed due to overflow`?

Reply:
I'm not sure what all those words mean in concert: Checked add to what? Overflow of what?

Reply:
This is an error for a different PR. It has been removed.